src/thunderbots/software/ai/hl/stp/tactic/receiver_tactic.cpp

```diff
@@ -0,0 +1,172 @@
+        Angle ball_robot_angle =
+            (ball_velocity.orientation() - robot_to_ball.orientation()).angleMod();
+
+        while(ball_robot_angle.toDegrees() < 90 || ball.velocity().len() < 0.5)
+        {
+            // Figure out the closest point on the balls trajectory to the robot
+            Point closest_ball_pos = closestPointOnLine(
+                robot->position(), ball.position(),
+                ball.estimatePositionAtFutureTime(Duration::fromSeconds(0.1)));
+            Ray shot(closest_ball_pos, best_shot_target - closest_ball_pos);
+
+            // Determine the best position to be in
+            Angle ideal_orientation      = getOneTimeShotDirection(shot, ball);
+            Vector ideal_orientation_vec = Vector::createFromAngle(
+                ideal_orientation);  // Vector(ideal_orientation.cos(),
+                                     // ideal_orientation.sin());
+
+            // The best position is determined such that the robot stays in the ideal
+            // orientation, but moves forwards/backwards so that the ball hits the kicker,
+            // rather then the center of the robot
+            Point ideal_position = closest_ball_pos - ideal_orientation_vec.norm(
+                                                          DIST_TO_FRONT_OF_ROBOT_METERS +
+                                                          BALL_MAX_RADIUS_METERS);
+
+
+            yield(move_action.updateStateAndGetNextIntent(
+                *robot, ideal_position, ideal_orientation, 0, false, true));
+
+            // Calculations to check for termination conditions
+            ball_velocity = ball.velocity();
+            robot_to_ball = ball.position() - robot->position();
+            ball_robot_angle =
+                (ball_velocity.orientation() - robot_to_ball.orientation()).angleMod();
+        }
+    }
+    // If we can't shoot on the enemy goal, just try to receive the pass as cleanly as
+    // possible
+    else
+    {
+        while ((ball.position() - robot->position()).len() >
+               DIST_TO_FRONT_OF_ROBOT_METERS + 2*BALL_MAX_RADIUS_METERS)
+        {
+            Point ball_receive_pos = closestPointOnLine(
+                robot->position(), ball.position(),
+                ball.estimatePositionAtFutureTime(Duration::fromSeconds(0.1)));
+            Angle ball_receive_orientation =
+                (ball.position() - robot->position()).orientation();
+
+            yield(move_action.updateStateAndGetNextIntent(
+                *robot, ball_receive_pos, ball_receive_orientation, 0, true, false));
+        }
+    }
+}
+
+Angle ReceiverTactic::getOneTimeShotDirection(const Ray& shot, const Ball& ball)
+{
+    Point shot_vector = shot.getDirection();
+    Angle shot_dir    = shot.getDirection().orientation();
+    Point bot_vector  = shot_vector.norm();
+
+    Point ball_vel       = ball.velocity();
+    Point lateral_vel    = ball_vel - (ball_vel.dot(-bot_vector)) * (-bot_vector);
```

Comment:
This is the component of the ball's velocity that's lateral/perpendicular to the shot vector, right?

I trust the math is correct but a slightly more explicit way might be
```
Point lateral_vel = ball_vel.project(bot_vector.perp());
```
Untested, but this might work.

Reply:
Looks like it passes the tests, so should be good? We'll test this tmw as well.